src/resolve/local.ts

```diff
@@ -17,61 +13,35 @@ export default async function resolveLocal (spec: PackageSpec, opts: ResolveOpti
 
   if (dependencyPath.slice(-4) === '.tgz' || dependencyPath.slice(-7) === '.tar.gz') {
     const name = getTarballName(dependencyPath)
-    return {
+    const resolution: Resolution = {
+      type: 'tarball',
       id: createLocalPkgId(name, dependencyPath),
-      root: path.dirname(dependencyPath),
-      fetch: createLocalTarballFetcher({
-        tarball: dependencyPath
-      })
+      tarball: `file:${dependencyPath}`,
     }
+    return {resolution}
   }
 
   if (opts.linkLocal) {
```

Comment:
But it is an npm config. You can create an `.npmrc` file with `link-local = true` in the root of the repo.